proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -455,7 +456,7 @@ public void done(String result, Throwable error)
                 }
             }
         };
-
+        thisState = State.FINISHING;
```

Comment:
This doesn't make sense to me, shouldn't it be in finish() so its only done once the web request is complete? (and the offline mode bit can then be removed)

Reply:
We need to change the State there because the request is asyn and you could send another enc packet witch  would be handled if its sent to the server befor mojang responses

Reply:
Why do we need FINISHED and FINISHING then?

Reply:
Do you mean should set the State to finished directly? And remove it from the finish Methode?

Reply:
I think so. With this change isn't finished otherwise redundant?

Reply:
Yes it is is

Reply:
I will remove it soon